client/blocks/importer/wordpress/import-everything/pre-migration/migration-credentials-form.tsx

```diff
@@ -0,0 +1,188 @@
+import { useMigrateProvisionMutation } from 'calypso/data/site-migration/migrate-provision-mutation';
+import { recordTracksEvent } from 'calypso/state/analytics/actions';
+import { updateCredentials } from 'calypso/state/jetpack/credentials/actions';
+import getJetpackCredentialsUpdateError from 'calypso/state/selectors/get-jetpack-credentials-update-error';
+import getJetpackCredentialsUpdateStatus from 'calypso/state/selectors/get-jetpack-credentials-update-status';
+import { StartImportTrackingProps } from './types';
+
+interface Props {
+	sourceSiteId: number;
+	targetSiteId: number;
+	startImport: ( props?: StartImportTrackingProps ) => void;
+	selectedHost: string;
+	onChangeProtocol: ( protocol: 'ftp' | 'ssh' ) => void;
+}
+
+export const MigrationCredentialsForm: React.FunctionComponent< Props > = ( props ) => {
+	const { sourceSiteId, targetSiteId, startImport } = props;
+	const translate = useTranslate();
+	const dispatch = useDispatch();
+
+	const [ formState, setFormState ] = useState( INITIAL_FORM_STATE );
+	const [ formErrors, setFormErrors ] = useState( INITIAL_FORM_ERRORS );
+	const [ formMode, setFormMode ] = useState( FormMode.Password );
+	const [ hasMissingFields, setHasMissingFields ] = useState( false );
+
+	const formSubmissionStatus = useSelector(
+		( state ) =>
+			getJetpackCredentialsUpdateStatus( state, sourceSiteId ) as
+				| 'unsubmitted'
+				| 'pending'
+				| 'success'
+				| 'failed'
+	);
+
+	const isFormSubmissionPending = formSubmissionStatus === 'pending';
+	const formHasErrors = formErrors && Object.keys( formErrors ).length > 0;
+
+	useEffect( () => {
+		// Clear the hasMissingFields flag when there are no more errors.
+		if ( ! formHasErrors ) {
+			setHasMissingFields( false );
+		}
+	}, [ formErrors ] );
+
+	// validate changes to the credentials form
+	useEffect( () => {
+		const errors = validate( formState, formMode );
+
+		// Shorten the root user error message as the original message used in the form is too long.
+		if ( formState.user === 'root' ) {
+			errors.user = {
+				message: translate( "We can't accept credentials for the root user." ),
+				waitForInteraction: true,
+			};
+		}
+
+		setFormErrors( errors );
+	}, [ formState, formMode ] );
+
+	useEffect( () => {
+		props.onChangeProtocol( formState.protocol as 'ftp' | 'ssh' );
+	}, [ formState.protocol ] );
+
+	useEffect( () => {
+		if ( formSubmissionStatus === 'success' ) {
+			startImport( {
+				type: 'with-credentials',
+			} );
+		}
+	}, [ formSubmissionStatus ] );
+
+	const handleUpdateCredentials = () => {
+		if ( formHasErrors ) {
+			return;
+		}
+
+		const credentials = { ...formState };
+
+		if ( formMode === FormMode.Password ) {
+			credentials.kpri = '';
+		} else if ( formMode === FormMode.PrivateKey ) {
+			credentials.pass = '';
+		}
+
+		// @todo custom event for migration flow?
+		dispatch( recordTracksEvent( 'calypso_jetpack_advanced_credentials_flow_credentials_update' ) );
```

Comment:
Does this event name suppose to be updated?

Something: `calypso_site_migration_credentials_update`

Reply:
Done! :-)